lib/collection/src/collection/mod.rs

```diff
@@ -390,31 +390,6 @@ impl Collection {
             }
         }
 
-        // Try to request shard transfer if replicas on the current peer are dead
-        if state == ReplicaState::Dead && self.this_peer_id == peer_id {
-            let transfer_from = replica_set
-                .peers()
-                .into_iter()
-                .find(|(_, state)| state == &ReplicaState::Active)
-                .map(|(peer_id, _)| peer_id);
-            if let Some(transfer_from) = transfer_from {
-                self.request_shard_transfer(ShardTransfer {
-                    shard_id,
-                    from: transfer_from,
-                    to: self.this_peer_id,
-                    sync: true,
-                    // For automatic shard transfers, always select some default method from this point on
-                    method: Some(
-                        self.shared_storage_config
-                            .default_shard_transfer_method
-                            .unwrap_or_default(),
-                    ),
-                })
-            } else {
-                log::warn!("No alive replicas to recover shard {shard_id}");
-            }
-        }
-
```

Comment:
This removes logic for requesting transfers on dead shards in the `set_shard_replica_state` function here.

I believe this is fine, because we do the same check in `sync_local_state`.

In fact, keeping this logic here is unsupported in this stateless implementation.